tests/backends/test_awkward.py

```diff
@@ -26,97 +27,120 @@
 ) < packaging.version.Version("2.2.3")
 
 
-def test_dimension_conversion():
+def _assert(bool1, bool2, backend):
+    # for 'typetracer' backend we pass through (no-op)
+    # for 'cpu' we want to make sure the bool evaluates to True
+    if backend == "cpu":
+        if isinstance(bool1, ak.Array):
+            bool1 = bool1.tolist()
+        if isinstance(bool2, ak.Array):
+            bool2 = bool2.tolist()
+        else:
+            assert bool1 == bool2
+
+
+@pytest.mark.parametrize("backend", ["cpu", "typetracer"])
+def test_dimension_conversion(backend):
+    assert_backend = functools.partial(_assert, backend=backend)
+
     # 2D -> 3D
     vec = vector.Array(
         [
             [{"x": 1, "y": 1.1}, {"x": 2, "y": 2.1}],
             [],
-        ]
+        ],
+        backend=backend,
     )
+
     # test alias
-    assert ak.all(vec.to_3D(z=1).z == 1)
-    assert ak.all(vec.to_3D(eta=1).eta == 1)
-    assert ak.all(vec.to_3D(theta=1).theta == 1)
+    assert_backend(ak.all(vec.to_3D(z=1).z == 1), True)
```

Comment:
Wouldn't it be a lot clearer if these were just inside an `if backend == "cpu"` block? This really hides what's happening and that it's a simi-no-op? Though I guess it's still computing them, it's just not checking to see if it's true? Could it be equal to a `backend == "cpu"` constant or something?

Reply:
For the typetracer backend I want the expressions still be computed, but we can't compare it to numerical values, which is why the assertion is essentially a no-op. This is to make sure that the computation does not fail for the typetracer backend (without checking its result).

Reply:
We might want to compare the `ak.forms.Form` in the typetracer case, but that's kind of already checked generously in awkward-array itself - so not really useful. 
So this is just to make sure a vector computation does not fail for the typetracer backend.